### ISSUE TITLE
[1.15.2] FPS drops with despawn sphere (when standing far away)

### ISSUE THREAD

**Issue by Azahliil** [maruohon/minihud#31]
Hey,

I noticed significant frame rate drops (from 120 to 3) on 1.15.2 when looking straight up. Any other direction is fine, as far as I can tell.

I'm not using Vsync and my frame rate is capped at 120

Starting the game fresh without MaLiLib and MiniHud "fixes" that problem tho, so I guess it has nothing to do with Fabric itself. 

Would be awesome if you could take a look at that!
And if I can provide any furth information on that please let me know :)

**Comment by maruohon** [comment]
Simply looking straight up does nothing in my case, so yes I will need more information. Can you narrow it down to any feature or renderer being enabled?

![2020-01-28_12 53 01](https://user-images.githubusercontent.com/2331050/73257936-56e26480-41cd-11ea-9077-4ac5d5a2c3bd.png)

**Comment by Azahliil** [comment]
I think I can narrow it down, I have one despawn sphere on this world, which is 20k blocks away from my current position, and that seems to cause the issue.
So at least I have an easy workaround now :)

**Comment by JJSax** [comment]
I too am having this exact issue.  Merely turning off the despawn sphere if I'm extremely far away fixes my issue when looking up or down.  I've noticed that if I'm very far down in the world, like y=10 or below or so, looking down isn't a big problem, but looking up is.  The higher up I am, the worse looking down is.  

I'm not sure if it's related but if I'm super far away from the despawn sphere, sometimes if I look just the right way, it tries to render the sphere as if I'm fairly close to it.

**Comment by supersaiyansubtlety** [comment]
I can also confirm I get an FPS drop when there are distant despawn spheres. 

It is at least easy to work around by just disabling the shape, but sometimes I forget and waste time troubleshooting until I remember to disable the shape.

**Comment by Nomenator** [comment]
Yeah. Just figured how it works, was able to reproduce this effect reliably under the following conditions:

1) Despawn sphere must be outside of render distance. (Outside the long cartesian render distance - it must not render in the corner of the rendered chunks when turning 45 degrees)

2) Look directly in the direction of the despawn sphere.

3) Turn 90 degrees in any direction.

FPS drops dramatically.


Must be something to do with normals.
